### PR TITLE
rustc_const_eval: adopt let else in more places

### PR DESCRIPTION
**Pull request by est31**
Continuation of #89933, #91018, #91481, #93046, #93590, #94011.

I have extended my clippy lint to also recognize tuple passing and match statements. The diff caused by fixing it is way above 1 thousand lines. Thus, I split it up into multiple pull requests to make reviewing easier. This PR handles rustc_const_eval.

### CODE REVIEW

File: compiler/rustc_const_eval/src/interpret/eval_context.rs [hunk 1]
```diff
@@ -631,14 +631,10 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
...
-                let (unsized_size, unsized_align) =
-                    match self.size_and_align_of(metadata, &field)? {
-                        Some(size_and_align) => size_and_align,
-                        None => {
-                            // A field with an extern type. We don't know the actual dynamic size
-                            // or the alignment.
-                            return Ok(None);
-                        }
+                let Some((unsized_size, unsized_align)) = self.size_and_align_of(metadata, &field)? else {
+                        // A field with an extern type. We don't know the actual dynamic size
```
Comment (RalfJung) [inline compiler/rustc_const_eval/src/interpret/eval_context.rs:635]: Indentation seems off here?

Comment (est31) [reply compiler/rustc_const_eval/src/interpret/eval_context.rs]: Thanks for pointing that out. Rustfmt currently doesn't support `let else`. I aligned formatting manually but missed that spot I guess.

File: compiler/rustc_const_eval/src/interpret/memory.rs [hunk 2]
```diff
@@ -1043,18 +1037,18 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'mir, 'tcx, M> {
...
         let src_alloc = self.get_raw(src_alloc_id)?;
         let src_range = alloc_range(src_offset, size);
         M::memory_read(&self.extra, &src_alloc.extra, src.provenance, src_range)?;
         // We need the `dest` ptr for the next operation, so we get it now.
         // We already did the source checks and called the hooks so we are good to return early.
-        let (dest_alloc_id, dest_offset, dest) = match dest_parts {
-            None => return Ok(()), // Zero-sized *destiantion*.
-            Some(dest_ptr) => dest_ptr,
+        let Some((dest_alloc_id, dest_offset, dest)) = dest_parts else {
+            // Zero-sized *destiantion*.
```
Comment (lcnr) [inline compiler/rustc_const_eval/src/interpret/memory.rs:1050]:
```suggestion
            // Zero-sized *destination*.
```
existing typo

**Review by lcnr — COMMENTED** [review]
nit, then r=me

**Pull request merged**